source/tests/system/expect_macros.h

```diff
@@ -0,0 +1,47 @@
+#ifndef NIDEVICE_GRPC_TESTS_EXPECT_MACROS
+#define NIDEVICE_GRPC_TESTS_EXPECT_MACROS
+
+#define CHECK_ERROR(session)                                          \
+  if (1) {                                                            \
+    auto response = client::get_error(stub(), session);               \
+    EXPECT_EQ("", std::string(response.error_description().c_str())); \
+  }
+
+#define EXPECT_RESPONSE_SUCCESS(response) \
+  if (1) {                                \
+    EXPECT_EQ(0, (response).status());    \
+  }
+
+#define EXPECT_RESPONSE_WARNING(expected_warning, response) \
+  if (1) {                                                  \
+    EXPECT_GT(expected_warning, 0);                         \
+    EXPECT_EQ(expected_warning, response.status());         \
+  }
+
+#define EXPECT_RESPONSE_ERROR(expected_error, response) \
+  if (1) {                                              \
+    EXPECT_LT(expected_error, 0);                       \
+    EXPECT_EQ(expected_error, response.status());       \
+  }
+
+#define EXPECT_SUCCESS(session, response) \
+  if (1) {                                \
+    EXPECT_RESPONSE_SUCCESS(response)     \
+    CHECK_ERROR(session);                 \
+  }
+
+#define EXPECT_ERROR(expected_error, message_substring, session, response) \
+  if (1) {                                                                 \
+    EXPECT_RESPONSE_ERROR(expected_error, response)                        \
+    const auto error = client::get_error(stub(), session);                 \
```

Comment:
Worth noting: not all drivers will have a `get_error` with this signature or with a `get_error`. I'm OK with "cross that bridge when we get there". This change has quite a bit of immediate benefit. 

One trick we could do is codegen something consistent in the `client` APIs. Or require that something be defined in scope in the test class. Either is probably better than going for a crazy SFINAE switch statement. Or ... we could pull these into rfmx expect macros.

Reply:
Yeah, good point. Right now it's named like it's general but is really only intended at this point for RFmx. I like the idea of just pulling it out into rfmx expect macros. Renamed the header to "expect_rfmx_macros.h" instead which will at least indicate it's RFmx specific. Then when/if we want to generalize it or branch it off in the future for other tests we can.